standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -473,6 +505,18 @@ private boolean checkIfTableIsUsable(String tableName, boolean configValue) thro
     }
     return tableExists;
   }
+  
+  @Override
+  @RetrySemantics.ReadOnly
+  public DatabaseProduct getDatabaseProduct() {
```

Comment:
why do we need this? product is available in dataSourceWrapper

Reply:
Good catch! This is a remnant of an earlier approach, removed.